ПР15

Megrief opened the pull request:
Добавлена возможность создавать плейлисты и добавлять в них треки.
Созданы таблицы в БД для плейлистов и треков, добавленных хотя бы в один плейлист.

Megrief committed:
Ripple effect возвращён

Megrief committed:
Исправлены недочеты:
- Удалены лишние атрибуты
- Background указывается для каждого экрана отдельно
- Решена проблема с непрозрачным Background у элементов меню BottomNavigationView

Megrief committed:
Форматирование кода

Megrief committed:
Playlists work in process...

Megrief committed:
Playlists work in process...

Megrief committed:
Playlists:
- Доработан экран PlaylistsFragment
- Сверстан экран PlaylistCreationFragment
- Добавлена логика создания плейлиста
- Экран Аудиоплеера переработан с Активити на Фрагмент для удобства навигации
- Добавлена новая таблица в базу данных
- Реализована возможность выбора изображения из внешнего хранилища
- Оптимизация кода.

Megrief committed:
Playlists:
- Добавлен BottomSheetLayout на экран Аудиоплеера

Megrief committed:
Playlists:
- Доработка.

Megrief committed:
Playlists:
- Доработка.

Megrief committed:
Playlists:
- Доработка.

DmitrijEgorow commented on app/src/main/java/com/example/playlistmaker/ui/playlist_creation/PlaylistCreationFragment.kt:
```diff
@@ -0,0 +1,204 @@
+                lifecycleScope.launch(Dispatchers.IO) {
+                    viewModel.storePlaylist()
+                }
+                Toast.makeText(requireContext(), "Плейлист $savedName создан", Toast.LENGTH_SHORT).show()
```
Лучше string-res
Возникнут проблемы при локализации

DmitrijEgorow commented on app/src/main/java/com/example/playlistmaker/ui/playlist_creation/view_model/PlaylistCreationViewModel.kt:
```diff
@@ -0,0 +1,43 @@
+import java.io.File
+
+class PlaylistCreationViewModel(
+    private val storePlaylistInDb: StoreDataUseCase<Playlist>,
```
Лучше убрать Db здесь
Если мы захотим использовать другой data source (shared preferences или файл, например), то мы немного изменим только репозиторий, но название свойства будет уже вводить в заблуждение

DmitrijEgorow commented:
Тост в этом случае лишний


https://github.com/Megrief/PlaylistMaker/assets/36837067/c1a70ff1-969a-4bb7-b6ec-7c6cb4ffde03

DmitrijEgorow commented:
- [ ] Название, наверное, не должно быть на несколько строк
- [ ] Название и описание обрезаются при увеличении размера текста устройства

![Screenshot_20231106_155811](https://github.com/Megrief/PlaylistMaker/assets/36837067/f771c643-f4e6-4b31-a182-1a2ad6c8914b)

DmitrijEgorow commented:
Нижнее меню не должно появляться на этом экране

https://github.com/Megrief/PlaylistMaker/assets/36837067/4001a603-0c04-465d-bc6b-5ac0f8c36f24

Megrief committed:
Playlists:
- Исправлено некорректое отображение увеличенного текста
- Тост на экране аудиоплеер больше не показывается без необходимости
- Количество строк в TextInputField исправлено на 1
- Необходимость отображения BottomNavigationView теперь проверяется на каждом экране в методе onResume()

Megrief committed:
Переписаны репозитории.

Megrief committed:
Исправлена ошибка, при которой не обновлялось состояние плейлистов на экране PlaylistsFragment
